src/main/java/org/apache/sling/event/impl/jobs/queues/JobExecutionContextImpl.java

```diff
@@ -84,6 +84,22 @@ public void updateProgress(final long eta) {
         }
     }
 
+    @Override
+    public void setProperty(final String name, final Object value) {
+        if ( name == null ) {
+            throw new IllegalArgumentException("Name must not be null");
+        }
+        if ( value == null ) {
+            throw new IllegalArgumentException("Value must not be null");
+        }
+        if ( name.startsWith("slingevent:") || name.startsWith(":slingevent:")) {
+            throw new IllegalArgumentException("Property name must not start with slingevent: " + name);
```

Comment:
We should make the exception message more precise and also mention ":slingegent:" as not allowed.